src/Opdex.Platform.WebApi/Models/Requests/Transactions/TransactionFilterParameters.cs

```diff
@@ -23,17 +24,19 @@ public TransactionFilterParameters()
         /// <summary>
         /// Optional list of smart contract addresses to filter transactions by.
         /// </summary>
+        [NotNull]
         public IEnumerable<Address> Contracts { get; set; }
 
         /// <summary>
         /// Filter transactions based on event types included.
         /// </summary>
+        [NotNull]
         public IEnumerable<TransactionEventType> EventTypes { get; set; }
```

Comment:
Can be nullable

Reply:
OpenAPI 3.0 has definitions for both nullable and required. Nullable would mean you can pass null explicitly, for example in the json body. For reference types NSwag generates the doc with nullable true. Here `[NotNull]` is removing the nullable definition from the OpenAPI doc, because the API doesn't accept a null query string.

Reply:
I'll have to look into that one, that sounds incredibly wrong and misleading.

Reply:
Take a look at the top answer here https://stackoverflow.com/questions/45575493/what-does-required-in-openapi-really-mean

To give an example, `?cursor=null` would be treated as the string "null" by asp.net core. NSwag adds `nullable: true` to everything which is a reference type, it doesn't determine whether it is a query param or not.

Reply:
Yeah I see required != not null - confusing at first but makes sense